.travis.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+if (3 > $_SERVER['argc']) {
+    echo "Usage: branch dir1 dir2 ... dirN\n";
+    exit(1);
+}
+
+$dirs = $_SERVER['argv'];
+array_shift($dirs);
+$branch = array_shift($dirs);
+
+$packages = array();
+$flags = PHP_VERSION_ID >= 50400 ? JSON_PRETTY_PRINT | JSON_UNESCAPED_SLASHES | JSON_UNESCAPED_UNICODE : 0;
+
+foreach ($dirs as $dir) {
+    $package = json_decode(file_get_contents($dir.'/composer.json'));
+    $name = $package->name;
+
+    $package->repositories = array(array(
+        'type' => 'composer',
+        'url' => 'file://'.__DIR__.'/',
+    ));
+    file_put_contents($dir.'/composer.json', json_encode($package, $flags));
+    passthru("cd $dir && tar -cf package.tar --exclude='package.tar' *");
+
+    $package->version = $branch.'.x-dev';
+    $package->dist['type'] = 'tar';
+    $package->dist['url'] = 'file://'.__DIR__."/$dir/package.tar";
+
+    $packages[$name][$package->version] = $package;
+
+    $package = file_get_contents('https://packagist.org/packages/'.$name.'.json');
+    $package = json_decode($package);
+
+    foreach ($package->package->versions as $version => $package) {
```

Comment:
you should not overwrite the ``$package`` variable here IMO. It hurts readability